db/c.cc

```diff
@@ -1392,6 +1393,17 @@ char* rocksdb_get_cf_with_ts(rocksdb_t* db,
   return result;
 }
 
+char* rocksdb_get_db_identity(rocksdb_t* db) {
+  std::string tmp;
+  Status s = db->rep->GetDbIdentity(tmp);
+  if (s.ok()) {
+    // ID is an ASCII GUID and we want NUL terminated string.
```

Comment:
> ID is an ASCII GUID

Not sure it's always true as `Env::GenerateUniqueId()` may be overridden. Its API doc says it returns something human-readable, which implies there should be no intervening '\0's. Still, this doesn't appear to be enforced anywhere, and from a quick look I think an implementation violating that comment might just work everywhere, except here...

Would it be too ugly to follow the pattern set forth by `rocksdb_column_family_handle_get_name()`?

Reply:
Ah great, thanks! I've change `rocksdb_get_db_identity` to follow that pattern for binary data, and updated the test. Thanks for the review!